### PR TITLE
[WIP] Add PVC docs

### PR DESCRIPTION
**Pull request by abrennan89**
Fixes https://github.com/knative/docs/issues/5083

## Proposed Changes

- Moves services docs to resources section
- Adds PVC info for services

Direct preview link: https://deploy-preview-5507--knative.netlify.app/docs/serving/services/pvc/

### CODE REVIEW

File: docs/serving/services/pvc.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Permanent data storage
+
+You can provide permanent data storage for Knative Services by configuring `PersistentVolumeClaims` (PVCs).
```
Comment (evankanderson) [inline docs/serving/services/pvc.md:3]: Nit: I'd describe this as "shared filesystem storage".  Given the description below, you'd probably also want to highlight that your volume claim will need an `accessMode` of `ReadWriteMany` if you want read-write access.  If you want read-only shared data, you can set the `readOnly` property of the `volume` entry to `true` and use `ReadOnlyMany`.

`ReadWriteOnce` and `ReadWriteOncePod` will not work with Knative autoscaling.

Comment (abrennan89) [reply docs/serving/services/pvc.md:3]: Great feedback, thanks!

File: config/nav.yml [hunk 2]
```diff
@@ -95,6 +95,21 @@ nav:
     - Serving:
       - Knative Serving overview: serving/README.md
       - Resources:
+        - Services:
+          - About Knative Services: serving/services/README.md
+          - Creating a Service: serving/services/creating-services.md
+          - Configuring Services:
+            - Configuring private Services: serving/services/private-services.md
+            - Permanent data storage: serving/services/pvc.md
```
Comment (skonto) [inline config/nav.yml:103]: I would recommend something like: "Data Storage Access" or "Storage Access", storage can be non permanent. 
In any case to be precise: "A PersistentVolumeClaim (PVC) is a request for storage by a user" that is why I am more inclined to change this.
Unless we are going to split storage to permanent and ephemeral.
For the latter we support [empty dir ]( https://deploy-preview-5507--knative.netlify.app/docs/serving/configuration/feature-flags/#kubernetes-emptydir-volume) which is part of the ephemeral storage volume types [provided by K8s](https://kubernetes.io/docs/concepts/storage/ephemeral-volumes/#types-of-ephemeral-volumes).

File: docs/serving/services/pvc.md (hunk 1 above)
Comment (skonto) [reply docs/serving/services/pvc.md:3]: I also provided some naming alternatives above. Besides the access mode, depending on the volume type users may have to adjust ksvc security context wrt to file related properties eg. FSGroup.

**Pull request closed without merging**